lib/segment/src/index/sparse_index/sparse_vector_index.rs

```diff
@@ -334,7 +334,7 @@ impl<TInvertedIndex: InvertedIndex> SparseVectorIndex<TInvertedIndex> {
 
         let deleted_point_bitslice = vector_query_context
             .deleted_points()
-            .unwrap_or(id_tracker.deleted_point_bitslice());
+            .unwrap_or_else(|| id_tracker.deleted_point_bitslice());
```

Comment:
This probably is not necessary.

The function just returns a reference and we don't do any business logic.

Since we use LTO we can assume it yeets all the intermediate function calls.

However, since we use LTO I don't expect it to be any worse either because I expect it to also remove the closure.

I have a slight preference to keep the previous. Thoughts?

Reply:
One argument may be: this will prevent making it costly if we ever do add business logic in those functions in the future. But, I'm not sure if that will ever happen.